### PR TITLE
SRTP-KDF FS Preview

### PR DESCRIPTION
**Pull request by kaleb-himes**
# Description

Preview of changes to come for SRTP-KDF FS

### CODE REVIEW

**Comment by kaleb-himes** [comment]
NOTE: Has a dependency on FIPS repo PR #254 being merged before this PR can pass all tests involving fips-ready checks.

File: wolfcrypt/src/coding.c [hunk 1]
```diff
@@ -219,18 +219,18 @@ int Base64_Decode(const byte* in, word32 inLen, byte* out, word32* outLen)
 #ifndef BASE64_NO_TABLE
         if (e1 < BASE64_MIN || e2 < BASE64_MIN || e3 < BASE64_MIN ||
                                                               e4 < BASE64_MIN) {
-            WOLFSSL_MSG("Bad Base64 Decode data, too small");
+//            WOLFSSL_MSG("Bad Base64 Decode data, too small");
```
Comment (SparkiDev) [inline wolfcrypt/src/coding.c:222]: Remove '//'

**Review by SparkiDev — APPROVED** [review]

**Comment by kaleb-himes** [comment]
retest this please

**Comment by kaleb-himes** [comment]
Adding a note specifically for the wolfCrypt v6.0.0 module where EVP use-cases are now being fully exposed for consumption while FIPS is enabled (support for third-party libraries that expect EVP features).

Found multiple instances in the newly exposed EVP code paths for ECC where setting keysize to zero (0) in EVP EC key generation and curve_id to zero (0) also while calling one of the various features that ultimately invoked `wolfSSL_EC_KEY_generate_key()` with these 0 inputs would result in a default selection of weaker keys than desired in FIPS approved mode (112-bit or 192-bit). The changes in https://github.com/wolfSSL/wolfssl/pull/7295/commits/d553c9c2fc6b307c96b63f926bd683573bb0e074 effectively add 2 gates to wolfSSL proper and there is a 3rd gate in the fips sources (separately tracked sources) to guard against weak key use in default key generation when FIPS mode is enabled. The three gates are as follows:

Gate 1)  At the pk.c level ensures 256-bit keys are used by default when FIPS is enabled.
Gate 2) At fips.c level guards against cases where keysize is set but too low or keysize is zero (0) but a default curve_id of sufficient length is passed. It does allow a pass-through scenario when both keysize and curve_id are both set to zero (0)
Gate 3) Finally at the ecc.c level when the pass-through in fips.c has been taken (which can result in a default selection of 256-bit but can also result in less than desirable defaults) by setting both keysize and curve_id to zero (0) the [final gate](https://github.com/wolfSSL/wolfssl/pull/7295/commits/d553c9c2fc6b307c96b63f926bd683573bb0e074#diff-2d763a3496b29a2ab95941d5809670a4a13de8fa3a00e238170019fc2b40d5b7R1839-R1847) will kick back an error anytime a default curve is selected that is too weak for running in approved mode (224-bit is the shortest curve allowed in approved mode) .

**Comment by kaleb-himes** [comment]
Rebased with upstream master and force pushed.

**Comment by kaleb-himes** [comment]
retest this please

**Review by JacobBarthelmeh — APPROVED** [review]

**Pull request merged**